azerothcore/azerothcore-wotlk
# Core/Player: Interrupt flag capturing even if damage is absorbed.

## Changes Proposed:
-  Flag capturing will be cancelled even if damage is absorbed.


## Issues Addressed:

- Closes #9270
- Closes https://github.com/chromiecraft/chromiecraft/issues/1726

## SOURCE:

Original report.
Thanks to the author and the contributor to TrinityCore that originally wrote this fix: @zwerg
## Tests Performed:

- Tested on my local build on Windows 10, I made two toons and applied PW:S and attempted to capture the flag while dealing damage on another character.


## How to Test the Changes:


 Create two opposite faction toons on your local server, one should be for instance a priest.
 make sure to:
 .learn all my class
 .debug bg 
queue Arathi Basin

1. Apply PW:S on yourself as the priest and capture any flag
2. Deal damage to the priest on the other toon by applying a DOT or a direct damage spell
3. See how the flag capturing has been cancelled although the damage has been absorbed.



## Known Issues and TODO List:


- I have only tested PW:S absorb so far against DOTS and direct damage. There could be more instances where flag capturing has incorrect behavior.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
